papers/hendrik_heuer/heuer.rst

```diff
@@ -0,0 +1,223 @@
+
+t-SNE achieves this by minimizing the Kullback-Leibler divergence between the joint probabilities of the high-dimensional data and the low-dimensional representation. The Kullback-Leibler divergence measures the faithfulness with which a probability distribution q represents a probability distribution p by a discrete scalar and equals zero if the distributions are the same [Maa08]. The Kullback-Leibler divergence is minimized using the gradient descent method. In contrast to other Stochastic Neighbor Embedding methods that use Gaussian distributions, it uses a Student t-distribution.
+
+Implementation
+--------------
+
+The text comparison tool implements a workflow that consists of a Python tool for the back-end and a Javascript tool for the front-end. With the Python tool, a text is converted into a collection of two-dimensional word vectors. These are visualized using the Javascript front-end. With the Javascript front-end, the user can explore the word map and zoom in and out to investigate both the local and the global structure of the text sources. The Javascript front-end can be published online.
+
+The workflow of the tool includes the following four steps: 
+
+Pre-processing
+~~~~~~~~~~~~~~
+
+In the pre-processing step, all sentences are tokenized to extract single words. The tokenization is done using the Penn Treebank Tokenizer implemented in the Natural Language Processing Toolkit (NLTK) for Python [Bir09]. Alternatively, this could also be achieved with a regular expression.
+
+Using a hash map, all words are counted. Only unique words, i.e. the keys of the hash map, are taken into account for the dimensionality reduction. The 3000 most frequent English words according to a frequency list collected from Wikipedia are ignored to reduce the amount of data.
+
+Word representations
+~~~~~~~~~~~~~~~~~~~~
+
+For all unique non-frequent words, the word representation vectors are collected from the word2vec model from the gensim Python library [Řeh10]. Each word is represented by an N-dimensional vector (N=300). 
+
+.. code-block:: python
+
+   from gensim.models import Word2Vec
+
+   model = Word2Vec.load_word2vec_format( \
+    word_vectors_filename, binary=True )
+
+   for word in words:
+     if word in model:
+       print model[ word ]
```

Comment:
This (and other code blocks) have inconsistent spacing inside brackets and parentheses. I would encourage following PEP8.